src/main/resources/public/jsx/geo-dash-widget-editor.js

```diff
@@ -1,318 +1,344 @@
 import React from "react";
 import ReactDOM from "react-dom";
 import _ from "lodash";
-import RGL, { WidthProvider } from "react-grid-layout"
+import RGL, { WidthProvider } from "react-grid-layout";
 const ReactGridLayout = WidthProvider(RGL);
 
 
 class BasicLayout extends React.PureComponent{
-    static defaultProps = {
-        isDraggable: true,
-        isResizable: true,
-        className: "layout",
-        items: 0,
-        rowHeight: 300,
-        onLayoutChange: function() {},
-        cols: 12,
-        graphReducer: "Min"
-    }
-    constructor(props) {
-        super(props);
-        this.state = {  layout: [],
-            widgets: [ ],
-            imagery: [],
-            isEditing: false,
-            selectedWidgetType: -1,
-            selectedDataType: -1,
-            WidgetTitle: "",
-            imageCollection: "",
-            graphBand: "",
-            graphReducer: "Min",
-            imageParams: "",
-            dualLayer: false,
-            WidgetBaseMap: "osm",
-            startDate:"",
-            endDate:"",
-            startDate2:"",
-            endDate2:"",
-            widgetBands:"",
-            widgetMin:"",
-            widgetMax:"",
-            widgetCloudScore:"",
-            imageCollectionDual: "",
-            selectedDataTypeDual: "-1",
-            imageParamsDual: "",
-            startDateDual:"",
-            endDateDual:"",
-            widgetBandsDual:"",
-            widgetMinDual:"",
-            widgetMaxDual:"",
-            widgetCloudScoreDual:"",
-            FormReady: false,
-            wizardStep: 1
-
-        };
-    };
+     static defaultProps = {
+         theURI: window.location.origin + "/geo-dash",
+         isDraggable: true,
+         isResizable: true,
+         className: "layout",
+         items: 0,
+         rowHeight: 300,
+         onLayoutChange: function () {
+         },
+         cols: 12,
+         graphReducer: "Min"
+     };
+     static getParameterByName (name, url) {
+         if (!url) {
+             url = window.location.href;
+         }
+         url = decodeURIComponent(url);
+         name = name.replace(/[\[\]]/g, "\\$&");
+         const regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)");
+         const results = regex.exec(url);
+         if (!results) {
+             return null;
+         }
+         if (!results[2]) {
+             return "";
+         }
+         return decodeURIComponent(results[2].replace(/\+/g, " "));
+     }
+     static getGatewayUrl(widget, collectionName){
+         let url = "";
```

Comment:
You dont need a variable here.  Just return the url from within each if block

Reply:
return window.location.protocol + "//" + window.location.hostname + ":8888/image";